### PR TITLE
Feature/284/implement refresh pass handler

### PR DESCRIPTION
**Pull request by BohdanLytvynov**
## Describe your changes . Please include a summary of the changes and  also include relevant motivation and context
- ResetPassword link sender using email is ready.
- Unit tests for ResetPasswordRequestHandler were created.

### CODE REVIEW

File: Streetcode/Streetcode.BLL/DTO/Users/RestorePasswordDto.cs [hunk 1]
```diff
@@ -0,0 +1,8 @@
+﻿namespace Streetcode.BLL.DTO.Users;
+
+public class RestorePasswordDto
```
Comment (Shavronskyy) [inline Streetcode/Streetcode.BLL/DTO/Users/RestorePasswordDto.cs:3]: maybe we need to check password twice?
like password and confirm password

File: Streetcode/Streetcode.BLL/MediatR/Account/RestorePassword/RestorePasswordRequestHandler.cs [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+                _emailConfiguration.From,
+                "Streetcode Restore Password Request",
+                "You have requested the restore password action.\n" +
+                "Please use this link for Password Reset: \n" +
+                $"{url}\n\t <b>If you don't request Password Restore, than do nothing and check your Accout!</b>");
+
+            bool success = await _emailService.SendEmailAsync(msg);
+
+            if (!success)
+            {
```
Comment (metrondir) [inline Streetcode/Streetcode.BLL/MediatR/Account/RestorePassword/RestorePasswordRequestHandler.cs:88]: provide some message if is not success

File: Streetcode/Streetcode.BLL/MediatR/Account/RestorePassword/RestorePasswordRequestHandler.cs [hunk 3]
```diff
@@ -0,0 +1,95 @@
...
+                    "RestorePassword",
+                    "Account",
+                    new 
+                    {
+                    userId = user.Id, 
+                    token 
+                    },
+                    _httpContextAccessor.HttpContext!);
+
+            Message msg = new Message(
```
Comment (metrondir) [inline Streetcode/Streetcode.BLL/MediatR/Account/RestorePassword/RestorePasswordRequestHandler.cs:77]: maybe create private method for that in handler

File: Streetcode/Streetcode.BLL/MediatR/Account/RestorePassword/RestorePasswordRequestHandler.cs [hunk 4]
```diff
@@ -0,0 +1,95 @@
...
+            _emailService = emailService;
+            _emailConfiguration = emailConfiguration;
+            _httpContextAccessor = httpContextAccessor;
+        }
+
+        public async Task<Result<string>> Handle(RestorePasswordRequest request, CancellationToken cancellationToken)
+        {
+            var user = await _userManager.FindByEmailAsync(request.restPassDto.Email);
+
+            // User not found request for pass change aborted
```
Comment (metrondir) [inline Streetcode/Streetcode.BLL/MediatR/Account/RestorePassword/RestorePasswordRequestHandler.cs:46]: erase that

File: Streetcode/Streetcode.BLL/DTO/Users/RestorePasswordDto.cs (hunk 1 above)
Comment (BohdanLytvynov) [reply Streetcode/Streetcode.BLL/DTO/Users/RestorePasswordDto.cs]: I thought, that it was a task for frontend.

File: Streetcode/Streetcode.BLL/MediatR/Account/RestorePassword/RestorePasswordRequestHandler.cs (hunk 4 above)
Comment (BohdanLytvynov) [reply Streetcode/Streetcode.BLL/MediatR/Account/RestorePassword/RestorePasswordRequestHandler.cs:49]: What should be erased, comment?

**Review by Shavronskyy — APPROVED** [review]
LGTM

**Review by PavloIeremeiew — APPROVED** [review]

**Pull request merged**